### ISSUE TITLE
[no-restricted-imports] Allow configuring severity at the path level so both errors and warnings are possible 

### ISSUE THREAD

**Issue by dshook** [eslint/eslint#14061]
**The version of ESLint you are using.**
6.8.0

**The problem you want to solve.**
I want to be able to configure the `no-restricted-imports` rule to have some paths be errors and others only be warnings.  After looking at the docs, the source for the rule, and much searching I couldn't find any way to configure it to do that. I believe this is a broader configuration issue than just the `no-restricted-imports` but this is the current rule I'm running into the roadblock with.

In JSON configuration

```
    "no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "somepackage",
            "message": "I want to make this one an error"
          },
          {
            "name": "otherpackage",
            "message": "But this should only be a warning"
          },
        ],
      }
    ],
```

**Your take on the correct solution to problem.**
```
    "no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "somepackage",
            "message": "I want to make this one an error"
          },
          {
            "name": "otherpackage",
            "message": "But this should only be a warning. By providing a severity here it overrides the severity set above",
            "severity": "warn"
          },
        ],
      }
    ],
```

**Are you willing to submit a pull request to implement this change?**
No

**Comment by mdjermanovic** [comment]
Hi @dshook, thanks for the issue!

This request looks similar to https://github.com/eslint/eslint/issues/13831.

Supporting this would require a lot of changes in ESLint's design, for maybe only a few rules where it would be useful.

As a workaround, you can make a clone of the rule and then configure the two rules separately (e.g., the original rule as `"error"`, and its clone as `"warn"`).

**Comment by dshook** [comment]
Thanks for the quick reply!

Just to be clear, when you say clone you mean using one of the other packages (like  eslint-rule-extender or eslint-rule-composer mentioned in the other issue) right?  I can't just put another instance of the `no-restricted-imports` in the config since it's a key of the rules object.

**Comment by mdjermanovic** [comment]
Yes, it should be two different rules, seemingly unrelated to each other.

eslint-rule-extender or eslint-rule-composer are actually not necessary if we just want to clone a rule without any modifications. New rule can just "re-export" the existing one, so this would be all the code of the new rule:

```js
const eslint = require("eslint");

module.exports = new eslint.Linter().getRules().get("no-restricted-imports");
```

Now you can use this module as a custom rule with [--rulesdir](https://eslint.org/docs/user-guide/command-line-interface#-rulesdir). For example, the above code can be placed in `my-rules/no-restricted-imports-clone.js`, and then the two rules would be configured like this:

```js
  rules: {
    "no-restricted-imports": [
      "error",
      {
        "paths": [
          {
            "name": "somepackage",
            "message": "I want to make this one an error"
          }
        ],
      }
    ],
    "no-restricted-imports-clone": [
      "warn",
      {
        "paths": [
          {
            "name": "otherpackage",
            "message": "But this should only be a warning"
          },
        ],
      }
    ]
  },
```

ESLint would be run with `--rulesdir my-rules`.

Another option is to make a plugin. For an example of a plugin that doesn't have to be published, you can see our internal rules used for this project only:

https://github.com/eslint/eslint/blob/4cab165bf4e6e5e9f42a59a37a8ff2548c0af87d/package.json#L102

The plugin is in https://github.com/eslint/eslint/tree/master/tools/internal-rules

**Comment by kylecombes** [comment]
Thanks, this works great! The only thing I had to add is an `/* eslint-disable */` to the top of the rule file (otherwise it complained about not being able to find the rule it was defining).

**Comment by kylecombes** [comment]
Looks like configuring it as a plugin (following [this guide](https://dev.to/spukas/how-to-write-your-first-eslint-plugin-145)) worked.

**Comment by nzakas** [comment]
Resolved, so closing.

**Comment by BarbecueSilver** [comment]
I am having the same issue, that we want `warnings` as well as `errors` for imports.
Thanks for pointing out, how to work around this issue :+1:
